extensions/sql-migration/src/models/stateMachine.ts

```diff
@@ -196,15 +201,103 @@ export class MigrationStateModel implements Model, vscode.Disposable {
 				}
 			}
 		});
-
 		this._assessmentResults = {
 			issues: serverLevelAssessments,
 			databaseAssessments: databaseLevelAssessments
 		};
-
+		this.generateAssessmentTelemetry(startTime, endTime, connectionProfile);
 		return this._assessmentResults;
 	}
 
+	private async generateAssessmentTelemetry(startTime: number, endTime: number, connectionProfile: azdata.connection.ConnectionProfile): Promise<void> {
+		try {
+			this._assessmentResults.issues.forEach(i => {
+				sendSqlMigrationActionEvent(
+					TelemetryViews.MigrationWizardTargetSelectionPage,
+					TelemetryAction.ServerAssessmentIssues,
+					{
+						'sessionId': this._sessionId,
+						'ruleId': i.ruleId
+					},
+					{
+						'count': i.impactedObjects.length
+					}
+				);
+			});
+
+			const queryProvider = azdata.dataprotocol.getProvider<azdata.QueryProvider>(connectionProfile.providerId, azdata.DataProviderType.QueryProvider);
+			const query = 'SELECT name, compatibility_level FROM sys.databases; ';
+			const results = await queryProvider.runQueryAndReturn(await (azdata.connection.getUriForConnection(this.sourceConnectionId)), query);
+			const compatLevelHashMap: Map<string, string> = new Map();
+			results.rows.forEach(r => {
+				compatLevelHashMap.set(r[0].displayValue, r[1].displayValue);
+			});
+
+			sendSqlMigrationActionEvent(
+				TelemetryViews.MigrationWizardTargetSelectionPage,
+				TelemetryAction.ServerAssessment,
+				{
+					'sessionId': this._sessionId,
+					'tenantId': this._azureAccount.properties.tenants[0].id,
+					'hashedServerName': hashString((connectionProfile.serverName))
+				},
+				{
+					'issuesCount': this._assessmentResults.issues.length,
+					'warningsCount': this._assessmentResults.databaseAssessments.reduce((count, d) => count + d.issues.length, 0),
+					'duration': endTime - startTime,
+					'databaseCount': this._assessmentResults.databaseAssessments.length,
+					'startTime': startTime,
+					'endTime': endTime,
+				}
+			);
+
+			const databaseMap: Map<string, number> = new Map();
+
+			this._assessmentResults.databaseAssessments.forEach(d => {
+
+				sendSqlMigrationActionEvent(
+					TelemetryViews.MigrationWizardTargetSelectionPage,
+					TelemetryAction.DatabaseAssessment,
+					{
+						'sessionId': this._sessionId,
+						'hashedDatabaseName': hashString(d.name),
+						'compatibilityLevel': compatLevelHashMap.get(d.name)!,
+					},
+					{
+						'warningsCount': d.issues.length
+					}
+				);
+
+				d.issues.forEach(i => {
+					if (databaseMap.has(i.ruleId)) {
```

Comment:
A bit simpler would just be `databaseMap.set(i.ruleId, databaseMap.get(i.ruleId) ?? 0 + i.impactedObjects.length);` instead of the separate if/else checks